app/com/linkedin/drelephant/exceptions/AzkabanClient.java

```diff
@@ -0,0 +1,239 @@
+      }
+
+      String result = parseContent(response.getEntity().getContent());
+      try {
+        jsonObj = new JSONObject(result);
+        if (jsonObj.has("error")) {
+          throw new RuntimeException(jsonObj.get("error").toString());
+        }
+      } catch (JSONException e) {
+        e.printStackTrace();
+      }
+    } catch (ClientProtocolException e) {
+      e.printStackTrace();
+    } catch (IOException e) {
+      e.printStackTrace();
+    } catch (NoSuchAlgorithmException e) {
+      e.printStackTrace();
+    } catch (UnrecoverableKeyException e) {
+      e.printStackTrace();
+    } catch (KeyManagementException e) {
+      e.printStackTrace();
+    } catch (KeyStoreException e) {
+      e.printStackTrace();
+    } finally {
+      httpClient.getConnectionManager().shutdown();
+    }
+    return jsonObj;
+  }
+
+  private String parseContent(InputStream response)
+      throws IOException {
+    BufferedReader reader = null;
+    StringBuilder result = new StringBuilder();
+    try {
+      reader = new BufferedReader(new InputStreamReader(response));
+
+      String line = null;
+      while ((line = reader.readLine()) != null) {
+        result.append(line);
+      }
+      return result.toString();
+    } catch (IOException e) {
+      e.printStackTrace();
+    } finally {
+      if (reader != null) {
+        reader.close();
+      }
+    }
+    return result.toString();
+  }
+
+  /*
+  * Authenticates Dr. Elephant in Azkaban and sets the sessionId
+  * @param username
+  * @param password
+  */
+  public void azkabanLogin(String userName, String password) {
+    List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+    urlParameters.add(new BasicNameValuePair("action", "login"));
+    urlParameters.add(new BasicNameValuePair("username", userName));
+    urlParameters.add(new BasicNameValuePair("password", password));
+
+    try {
+      JSONObject jsonObject = fetchJson(urlParameters);
+
+      if (!jsonObject.has("session.id")) {
+        throw new RuntimeException("Login attempt failed. The session ID could not be obtained.");
+      }
+
+      this._sessionId = jsonObject.get("session.id").toString();
+    } catch (JSONException e) {
+      e.printStackTrace();
+    }
+  }
+
+
+  /*
+   * Returns the Azkaban flow log for given Azkaban execution url.
+   *
+   * @param offset Offset of log from the start
+   * @param length Maximum limit on length of log
+   * @return Azkaban flow log in the form of string
+   */
+
+  public String getAzkabanFlowLog(String offset, String length) {
+    List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+    urlParameters.add(new BasicNameValuePair("session.id", _sessionId));
+    urlParameters.add(new BasicNameValuePair("ajax", "fetchExecFlowLogs"));
+    urlParameters.add(new BasicNameValuePair("execid", _executionId));
+    urlParameters.add(new BasicNameValuePair("offset", offset));
+    urlParameters.add(new BasicNameValuePair("length", length));
```

Comment:
It indicates the limit on the number of character of logs and not response. So if the the logs exceeds the specified number of characters only partial logs logs will be retrieved. For now, I have set it to 99999999 characters.